### PR TITLE
fix: allow `ca`, `key` and `cert` will be string (regression)

### LINKED ISSUE #1674: Passing an SSL key or certificate as a raw string results in an exception
- Operating System: Windows 10
- Node Version: 11.1.0
- NPM Version: 6.4.1
- webpack Version: 4.27.1
- webpack-dev-server Version: 3.2.0

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

```ts
// startWebpackDevServer.ts
/** `ssl` contains the `key` and `cert` properties as raw strings, rather than filenames */
const { ssl } = getUserConfig();
httpsConfig = {
    https: {
        ...ssl,
    },
};

const server = new WebpackDevServer(
        compiler,
        Object.assign(
            httpsConfig,
            {
                port: 3000,
                inline: true,
                quiet: false,
                noInfo: false,
                headers: {
                    'Access-Control-Allow-Origin': '*',
                    'Access-Control-Allow-Credentials': 'true',
                },
                stats: 'errors-only',
                reporter: reporter,
            },
            config.devServer
        )
    );
```

### Expected Behavior
`webpack-dev-server` operates as it did prior to 3.2.0, without throwing an exception

### Actual Behavior
```
(node:15556) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, lstat '-----BEGIN RSA PRIVATE KEY-----
<<REDACTED>>
-----END RSA PRIVATE KEY-----
'
    at Object.lstatSync (fs.js:861:3)
    at new Server (D:\vso\client-web\node_modules\owa-build\node_modules\webpack-dev-server\lib\Server.js:573:38)
```

### For Bugs; How can we reproduce the behavior?
Pass the raw SSL key and/or cert, rather than a Buffer or filepath, to the https config

### PR DESCRIPTION
**Pull request by evilebottnawi**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes

### Motivation / Use-Case

fixes #1674

### Breaking Changes

No

### Additional Info

It is regression, just not `pfx` can't be string due it is binary format

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -610,7 +623,7 @@ class Server {
           const pems = createCertificate(attrs);
 
           fs.writeFileSync(certPath, pems.private + pems.cert, {
-            encoding: 'utf-8',
+            encoding: 'utf8',
```
Comment (evilebottnawi) [inline lib/Server.js]: `utf8` is right value https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options

File: lib/Server.js [hunk 2]
```diff
@@ -570,8 +570,21 @@ class Server {
...
+
+          try {
+            stats = fs.lstatSync(value).isFile();
+          } catch (error) {
+            // ignore error
+          }
+
+          if (stats) {
+            // It is file
+            options.https[property] = fs.readFileSync(path.resolve(value));
```
Comment (alan-agius4) [inline lib/Server.js]: Shouldn’t this have utf8 encoding as well?

Comment (evilebottnawi) [reply lib/Server.js]: @alan-agius4 hm, maybe, can you provide example?

Comment (evilebottnawi) [reply lib/Server.js]: No need 
https://github.com/nodejs/node/blob/master/lib/fs.js#L392

`buffer.toString` :
```
encoding <string> The character encoding to use. Default: 'utf8'.

buf.toString([encoding[, start[, end]]])
```

Comment (evilebottnawi) [reply lib/Server.js]: So default is `utf8`

Comment (alan-agius4) [reply lib/Server.js]: thanks for clarifying

**Pull request merged**